internal/poolutil/pool.go

```diff
@@ -0,0 +1,44 @@
+// Copyright 2024 Nutanix. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package poolutil implements utility functions to manage a pool of IP addresses.
+package poolutil
+
+import (
+	"math"
+	"math/big"
+
+	"go4.org/netipx"
+)
+
+// IPSetCount returns the number of IPs contained in the given IPSet.
+// This function returns type int64, which is smaller than
+// the possible size of a range, which could be 2^128 IPs.
+// When an IPSet's count is would be larger than max int64, math.MaxInt64
+// is returned instead.
```

Comment:
This behavior seems confusing to me, should we error instead if total is larger than MaxInt64 or maybe use a type that can hold larger numbers?

Reply:
I've changed it to return an error, but I don't think anyone will ever hit this limit!